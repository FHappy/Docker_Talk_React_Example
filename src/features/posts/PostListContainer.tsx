import React from "react";
import { useSelector } from "react-redux";
import { Col, Row } from "react-bootstrap";

import PostList from "./PostList";
import { selectPostsByCurrentUser, selectPostsByOtherUsers } from "./postSlice";

export interface PostListContainerProps {}

const PostListContainer: React.FC<PostListContainerProps> = () => {
  const postsByCurrentUser = useSelector(selectPostsByCurrentUser);
  const postsByOtherUsers = useSelector(selectPostsByOtherUsers);

  return (
    <div style={{ padding: "2rem 0" }}>
      <Row>
        <Col xs={12} lg={6}>
          <PostList posts={postsByCurrentUser} title={"Your Posts"} />
        </Col>

        <Col xs={12} lg={6}>
          <PostList posts={postsByOtherUsers} title={"Posts by Other Users"} />
        </Col>
      </Row>
    </div>
  );
};

export default PostListContainer;
