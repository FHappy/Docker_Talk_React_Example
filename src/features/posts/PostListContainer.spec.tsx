import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { Provider } from "react-redux";

import store from "app/store";
import PostListContainer from "./PostListContainer";

describe("Rendering the container for both PostList components", () => {
  it("Should render the PostListContainer component correctly", () => {
    const wrapper: ShallowWrapper = shallow(
      <Provider store={store}>
        <PostListContainer />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
