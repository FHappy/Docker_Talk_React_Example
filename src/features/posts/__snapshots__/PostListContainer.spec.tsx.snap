// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the container for both PostList components Should render the PostListContainer component correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <PostListContainer />
</ContextProvider>
`;
